mm/ktsan/stack_depot.c

```diff
@@ -0,0 +1,132 @@
+#include "ktsan.h"
+
+#include <linux/jhash.h>
+#include <linux/kernel.h>
+
+#define KT_STACK_HASH_SEED 0x9747b28c
+
+/* Only available during early boot. */
+void __init kt_stack_depot_init(kt_stack_depot_t *depot,
+			unsigned part_count, unsigned stack_count)
+{
+	unsigned i;
+
+	depot->part_count = part_count;
+	depot->stack_count = stack_count;
+
+	kt_cache_init(&depot->part_cache,
+			sizeof(*depot->parts) * part_count, 1);
+	depot->parts = kt_cache_alloc(&depot->part_cache);
```

Comment:
this looks weird that you create a cache to allocate a single object
just embed parts into kt_stack_depot_t, it is like a megabyte